pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -61,6 +62,27 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		cr.Spec.Version = ""
 	}
 
+	// if the secret is the only credential present
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminUser) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
+		creds := constants.KIE_ADMIN_CREDENTIALS_SECRET
+		if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 {
+			creds = cr.Spec.CommonConfig.SecretAdminCredentials
+		}
```

Comment:
The second if is not needed to be run again, we can skip the second if and also setting default creds
```suggestion
	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminUser) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
		creds = cr.Spec.CommonConfig.SecretAdminCredentials
```

Reply:
ok